plugins/containers-plugin/src/containers-tree.ts

```diff
@@ -0,0 +1,115 @@
+/*********************************************************************
+ * Copyright (c) 2018 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import * as theia from '@theia/plugin';
+import { MachinesService } from "./machine-service";
+
+const machines: Map<string, string[]> = new Map<string, string[]>();
+
+export class ContainersTree {
+
+    constructor() {
+        const treeDataProvider = new TestDataProvider();
+
+        theia.window.createTreeView('containers', { treeDataProvider });
+
+        this.initTreeData().then(() => {
+            treeDataProvider.sendDataChanged();
+        });
+    }
+
+    async initTreeData(): Promise<void> {
+        const machinesService = new MachinesService();
+
+        await machinesService.updateMachines();
+
+        machinesService.machines.forEach(machine => {
+            const machineName = machine.machineName;
+            const machineProp: Array<string> = [];
+
+            if (machine.status) {
+                machineProp.push(machine.status);
+            }
+
+            const commands = ['terminal'];
+            commands.forEach((command: string) => {
+                machineProp.push(`${command}__${machineName}`);
+            });
+
+            const servers = machine.servers;
+            if (servers) {
+                Object.keys(servers).forEach((serverName: string) => {
+                    const server = servers[serverName];
+                    if (server && server.url) {
+                        machineProp.push(` [${serverName}](${server.url})`);
+                    } else {
+                        machineProp.push(serverName);
+                    }
+                });
+            }
+
+            machines.set(machineName, machineProp);
+        });
+    }
+
+}
+
+export class TestDataProvider implements theia.TreeDataProvider<string> {
+
+    private onDidChangeTreeDataEmitter: theia.EventEmitter<any> = new theia.EventEmitter<any>();
+    readonly onDidChangeTreeData: theia.Event<any> = this.onDidChangeTreeDataEmitter.event;
```

Comment:
Maybe `void` should be better instead of `any`

Reply:
done